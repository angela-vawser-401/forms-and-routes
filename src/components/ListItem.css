.ListItem {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 20px;

  li {
    width: 300px; 
    height: 250px;
    border: 3px solid black;
    background: black;
    padding: 0;
  }

  li:hover {
    transition: transform 100ms ease-in-out;
    box-shadow: 3px 3px 4px darkgray;
    transform: translate3d(-1.5px, 1.5px, 0);
  }

  div {
    max-height: 200px;
    width: 300px;
    object-fit: contain;
  }

  img {
    width: auto;
    max-width: 300px;
    height:100%;
    max-height: 200px;
    object-fit: contain;
  }

  h2 {
    text-align: center;
    text-justify: center;
    color: whitesmoke;
    font-weight: 700;
    text-transform: uppercase;
    text-decoration: none;
    font-family: sans-serif;
    height: 55px;
  }
}