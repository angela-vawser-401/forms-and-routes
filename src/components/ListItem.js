import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './ListItem.css';


class ListItem extends Component {

  static propTypes = {
    _id: PropTypes.string,
    name: PropTypes.string,
    photo: PropTypes.string,
    history: PropTypes.object
  }


  render() {
    const characterUrl = `/character/${this.props._id}`;
    return (
      <Link to={characterUrl}>
        <section className={styles.ListItem}>
          <li key={this.props._id}>
            <img src={this.props.photo} alt={this.props._id} />
            <h2>{this.props.name}</h2>
          </li>
        </section>
      </Link>
    );
  }
}

export default ListItem;
