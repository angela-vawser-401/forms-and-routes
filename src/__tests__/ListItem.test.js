import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../components/ListItem';

describe('ListItem component', () => {
  it('renders ListItem', () => {
    const item = {
      _id: '1234',
      name: 'Bart Simpson',
      photoUrl: 'eatmyshorts.jpg'
    };
    const wrapper = shallow(<ListItem
      {...item} />);
    expect(wrapper).toMatchSnapshot();
  });
});
