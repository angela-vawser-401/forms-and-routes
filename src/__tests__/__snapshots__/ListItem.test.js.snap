// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders ListItem 1`] = `
<Link
  to="/character/1234"
>
  <section
    className="ListItem"
  >
    <li
      key="1234"
    >
      <img
        alt="1234"
      />
      <h2>
        Bart Simpson
      </h2>
    </li>
  </section>
</Link>
`;
